display_list/effects/dl_image_filter.h

```diff
@@ -742,6 +751,101 @@ class DlLocalMatrixImageFilter final : public DlImageFilter {
   std::shared_ptr<const DlImageFilter> image_filter_;
 };
 
+class DlRuntimeEffectImageFilter final : public DlImageFilter {
+ public:
+  explicit DlRuntimeEffectImageFilter(
```

Comment:
There are a few things I'd like to clean up here as well (and the other files in this directory). (There is a .cc file for each, we just don't use it much.)

- Move more code to the .cc file as @gaaclarke is suggesting.
- Deskiafying the geometry and other simple objects, naturally.
- Disallow stack allocation and make them all shared pointers. It's great for unit tests where we don't care about performance, but causes extra copies in the primary code. We can still inline them into the DL records with private constructors, but if any DlOpReceiver code wants to save them "for later", it then needs to deal with reconstructing a shared version. :( Still, the amount of annoyance to deal with stack allocation vs shared_ptr feels like it should be cleaned up. For example, if you ever put it into a DlPaint, we have to re-share it when we should just keep a (shared) copy of the original.
- Having them be shared pointers means we never construct a NOP filter because it would notice that it was NOP and return a null instead of a constructed object that has an identity (am I useful?) crisis.
- Have different files for the different types??? This would explode the source files in this directory, but isolate the implementations. There would still be a main "dl_image_filter.h" that included the others as it needs to support the type and `asFoo()` mechanisms, but we wouldn't have one giant regurgitation of code in the main file.
- Use more vectors instead of count/ptr pairs.